litestar/openapi/controller.py

```diff
@@ -1,31 +1,79 @@
 from __future__ import annotations
```

Comment:
Given that this entire controller is documented, I might have gone too hard on the changes. However, using DI and guards etc has really tidied this up (who'd have thought?!😀), and I really just wanted to see what it looked like. Maybe we could have an alternate controller that is registered by the plugin based on feature flag and just leave this one as is.